VContainer/Assets/VContainer/Runtime/Container.cs

```diff
@@ -55,17 +56,20 @@ public sealed class ScopedContainer : IScopedObjectResolver
         readonly ConcurrentDictionary<Registration, Lazy<object>> sharedInstances = new ConcurrentDictionary<Registration, Lazy<object>>();
         readonly CompositeDisposable disposables = new CompositeDisposable();
         readonly Func<Registration, Lazy<object>> createInstance;
+        readonly IEnumerable<Action<IObjectResolver>> disposeCallbacks;
 
         internal ScopedContainer(
             Registry registry,
             IObjectResolver root,
             IScopedObjectResolver parent = null,
-            object applicationOrigin = null)
+            object applicationOrigin = null,
+            IEnumerable<Action<IObjectResolver>> disposeCallbacks = null)
```

Comment:
I also noticed recently that if you only keep the Action<IObjectResolver> disposeCallback type perhaps,

```cs
disposeCallback += callback
```

I think it should be possible to register more than one callback, as in
You can verify this if you like.